2021q3/stackgap.adoc

```diff
@@ -0,0 +1,25 @@
+=== Stack gap handling improvements
+
+Contact: Dawid Gorecki <dgr@semihalf.com> +
+Contact: Marcin Wojtas <mw@semihalf.com>
+
+Stack gap is a feature, which randomizes the stack address by creating a random sized gap between top of stack and strings area.
+The current implementation of this mitigation can cause issues for some applications e.g. Firefox https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=239873[PR239873].
+Until now the only workaround for this problem was disabling the stack gap for those programs, as is done for https://cgit.freebsd.org/src/commit/usr.sbin/ntp/ntpd?id=af949c590bd8a00a5973b5875d7e0fa6832ea64a[ntpd].
+
+Semihalf team has been working on fixing the root cause of a stack gap related problems.
+
+One of the issues could be observed when setting the stack limit to a low value with setrlimit(2).
+Since the stack gap size can be significant, and the process had no knowledge of how big stack gap is, this caused programs to abort immediately after returning from syscall due to stack extending past limit.
```

Comment:
s/big stack gap/large the stack gap/ (grammar + "large" is a more natural word choice)
"from a syscall" or "from syscalls"
s/to stack/to the stack/
s/past limit/past the limit/